File: scripts/SpatialCells/linear_walk.m

function [Xpos,Ypos] = linear_walk(W,tend,dt)

nsteps = floor(tend/dt);
velocity = rand()/2;

Xpos = zeros(nsteps,1);
Ypos = zeros(nsteps,1);
headDir = zeros(nsteps,1)';

Xpos(1) = 0;
Ypos(1) = 0;

for i = 2:nsteps
    % max acceleration is .1 cm/ms^2
    dv = max(min(normrnd(0,.05),.1),-.1); 
    
    % max velocity is .05 cm/ms
    velocity = min(max(velocity + dv,0),0.05) * dt;
        
    Xpos(i) = Xpos(i-1)+cos(headDir(i-1))*velocity; 
    Ypos(i) = Ypos(i-1);

end

File: unused/SpatialCells/linear_walk.py
import math
import numpy as np

def linear_walk(tend,dt):

    np.random.seed(seed=21)

    ## max acceleration is .1 cm/ms^2
    max_accel = 0.1
    ## max velocity is .05 cm/ms
    max_vel   = 0.05
    
    nsteps   = int(math.floor(tend/dt))
    velocity = np.random.rand()/2

    Xpos = np.zeros(nsteps,)
    Ypos = np.zeros(nsteps,)
    headDir = np.zeros(nsteps,)
    
    for i in xrange(1,nsteps):
        a  = np.random.randn()+0.05
        dv = max(min(a,max_accel),-max_accel); 
        velocity = min(max(velocity + dv,0.0),max_vel) * dt
        
        Xpos[i] = Xpos[i-1] + np.cos(headDir[i-1]) * velocity 
        Ypos[i] = Ypos[i-1]

    return (Xpos,Ypos)
 
